### PR TITLE
Gi clipmaps fix latency and crashes

### PR DESCRIPTION
**Pull request by e2002e**
Fixes crashes when allocating mipmaps for 3D images.
I think that the pc runs out of ram because mipmapping for 3d images is, correct my if i'm wrong, 8 times the size of the original image. And anisotropy, precomputed direction, and clipmaps are stored in these images by extending their sizes so we don't use mipmaps.

@luboslenco this is kind of crucial that you merge this one, I'm opening an 'overhaul' pull request for armory but even with the current code the patch is needed.

### CODE REVIEW

File: Sources/iron/RenderPath.hx [hunk 1]
```diff
@@ -687,17 +689,15 @@ class RenderPath {
 		if (height < 1) height = 1;
 		if (t.depth != null && t.depth > 1) { // 3D texture
 			// Image only
-			var img = Image.create3D(width, height, depth,
-				t.format != null ? getTextureFormat(t.format) : TextureFormat.RGBA32);
-			if (t.mipmaps) img.generateMipmaps(1000); // Allocate mipmaps
```
Comment (luboslenco) [inline Sources/iron/RenderPath.hx:692]: Hm why is the `generateMipmaps()` call removed?

Comment (e2002e) [reply Sources/iron/RenderPath.hx:692]: Because we don't want to use mipmaps with 3D images.
It is unneeded and what we would gain in speed when sampling we loose in RAM.
I had done some tests with mipmaps and anyways it's not looking good, the original module as it's in WickedEngine doesn't use mipmaps.

Comment (luboslenco) [reply Sources/iron/RenderPath.hx:692]: Why not just set `t.mipmaps` to `false` then? Right now you removed the ability to generate mipmaps of 3d images completely.

Comment (e2002e) [reply Sources/iron/RenderPath.hx:692]: I'm not certain but I don't think there was any mipmaps generating system in iron for 3D images before.
The mipmaps where generated in the renderpath.
But yes, if you prefer that it is done this way then OK.

Comment (luboslenco) [reply Sources/iron/RenderPath.hx:692]: > I'm not certain but I don't think there was any mipmaps generating system in iron for 3D images before.

I checked some random old revision and looks like the `if (t.mipmaps) img.generateMipmaps(1000);` has been there for a long time:

https://github.com/armory3d/iron/blob/f263fa92b9a3d7759353479f167d8e5a7d6a4408/Sources/iron/RenderPath.hx#L658

File: Sources/iron/RenderPath.hx [hunk 2]
```diff
@@ -689,14 +689,18 @@ class RenderPath {
...
+				img.generateMipmaps(1000); // Allocate mipmaps
+			return img;
 		}
 		else { // 2D texture
 			if (t.is_image != null && t.is_image) { // Image
 				var img = Image.create(width, height,
 					t.format != null ? getTextureFormat(t.format) : TextureFormat.RGBA32);
 				if (t.mipmaps)
-					img.generateMipmaps(16); // Allocate mipmaps
+					img.generateMipmaps(1000); // Allocate mipmaps
```
Comment (luboslenco) [inline Sources/iron/RenderPath.hx:703]: It is really confusing to review it like this, the value was originally 1000, then was changed to 16, and now it's back to 1000 in an unrelated commit. :)

Comment (e2002e) [reply Sources/iron/RenderPath.hx:703]: I know, actually I was trying to optimize the mipmaps before disabling it.
Let it be a thousand and anyways for voxels we don't use mipmaps.
